### PR TITLE
RAC-4819 - convert decommission nodes script to FIT

### PR DESCRIPTION
**Pull request by larry-dean**
* Created fit version of decommission test

### CODE REVIEW

**Comment by larry-dean** [comment]
@hohene @johren @BrentHiggins

**Comment by BrentHiggins** [comment]
+1

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 1]
```diff
@@ -0,0 +1,116 @@
+'''
+Copyright 2016, EMC, Inc.
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Copyright (c) 2017 Dell Inc. or its subsidiaries. All Rights Reserved

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+                log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+                log.info_5(" Workflow completed at time: " + str(fit_common.time.time()))
+                return True
+            else:
+                log.error(" Workflow failed: status: %s text: %s", result['json']['status'], result['text'])
+                return False
+        log.error(" Workflow Timeout: " + result['text'])
+        return False
+
+    # @depends(after=test01_node_check)
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Can this depends line be removed?

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    def __wait_for_workflow_complete(self, instanceid, start_time, waittime=900, cycle=30):
+        log.info_5(" Workflow started at time: %f", start_time)
+        while time.time() - start_time < waittime:
+            result = fit_common.rackhdapi("/api/2.0/workflows/" + instanceid)
+            if result['status'] != 200:
+                log.error(" HTTP error: " + result['text'])
+                return False
+            if result['json']['status'] in ['running', 'pending']:
+                log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+                fit_common.time.sleep(cycle)
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Please remove the fit_common here.  We should use this directly from the time import.

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+            elif result['json']['status'] == 'succeeded':
+                log.info_5("{} workflow status: {}".format(result['json']['injectableName'], result['json']['status']))
+                log.info_5(" Workflow completed at time: " + str(fit_common.time.time()))
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Remove fit_common here, too.

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        # launch workflow
+        workflowid = None
+        result = fit_common.rackhdapi('/api/2.0/nodes/' +
+                                      self.__class__.__NODE +
+                                      '/workflows',
+                                      action='post', payload=DECOMMISSION_PAYLOAD)
+        if result['status'] == 201:
+            # workflow running
+            log.info_5(" InstanceID: " + result['json']['instanceId'])
+            log.info_5(" Payload: " + fit_common.json.dumps(DECOMMISSION_PAYLOAD))
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Please import json directly and not use fit_common here.

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 6]
```diff
@@ -0,0 +1,116 @@
...
+            workflowid = result['json']['instanceId']
+        else:
+            # workflow failed with response code
+            log.error(" InstanceID: " + result['text'])
+            log.error(" Payload: " + fit_common.json.dumps(DECOMMISSION_PAYLOAD))
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Same here.

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+        # launch workflow
+        workflowid = None
+        result = fit_common.rackhdapi('/api/2.0/nodes/' +
+                                      self.__class__.__NODE +
+                                      '/workflows',
+                                      action='post', payload=VALIDATE_PAYLOAD)
+        if result['status'] == 201:
+            # workflow running
+            log.info_5(" InstanceID: " + result['json']['instanceId'])
+            log.info_5(" Payload: " + fit_common.json.dumps(VALIDATE_PAYLOAD))
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: Remove fit_common

File: test/tests/rackhd20/test_rackhd20_api_decommission_node.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+            workflowid = result['json']['instanceId']
+        else:
+            # workflow failed with response code
+            log.error(" InstanceID: " + result['text'])
+            log.error(" Payload: " + fit_common.json.dumps(VALIDATE_PAYLOAD))
```
Comment (johren) [inline test/tests/rackhd20/test_rackhd20_api_decommission_node.py]: remove fit_common

**Comment by larry-dean** [comment]
Pushed up changes suggested by PR review

**Comment by larry-dean** [comment]
test this please

**Pull request merged**